onnxruntime/contrib_ops/cpu/quantization/gather_block_quantized.cc

```diff
@@ -186,21 +218,44 @@ Status GatherBlockQuantized<T1, Tind>::CopyDataAndDequantize(const T1* data_ptr,
     int64_t output_idx = output_idx_base;
     int64_t data_idx = data_idx_base;
     for (int64_t i = 0; i < gather_block; ++i, ++output_idx, ++data_idx) {
-      auto data_val = GetDataElement(data_ptr, data_idx);
+      int32_t data_val;
```

Comment:
Shall we initialize this variable?

Reply:
It is fine. Every path in lines 222-230 below will set a value.